testsupport/wait/host.go

```diff
@@ -919,18 +919,61 @@ func (a *HostAwaitility) WaitForNSTemplateTierAndCheckTemplates(t *testing.T, na
 		if ns.TemplateRef == "" {
 			return nil, fmt.Errorf("missing 'templateRef' in namespace #%d in NSTemplateTier '%s'", i, tier.Name)
 		}
-		if _, err := a.WaitForTierTemplate(t, ns.TemplateRef); err != nil {
+		tierTemplateNamespaces, err := a.WaitForTierTemplate(t, ns.TemplateRef)
+		if err != nil {
 			return nil, err
 		}
+		// if the tier template supports Tier Template Revisions then let's check those
+		if tierTemplateNamespaces.Spec.TemplateObjects != nil {
+			ttrName, found := tier.Status.Revisions[tierTemplateNamespaces.GetName()]
+			if !found {
+				return nil, fmt.Errorf("missing revision for TierTemplate %s in NSTemplateTier '%s'", tierTemplateNamespaces.GetName(), tier.Name)
+			}
+			if _, err := a.WaitForTierTemplateRevision(t, ttrName); err != nil {
+				return nil, err
+			}
+		}
 	}
 	if tier.Spec.ClusterResources != nil {
 		if tier.Spec.ClusterResources.TemplateRef == "" {
 			return nil, fmt.Errorf("missing 'templateRef' for the cluster resources in NSTemplateTier '%s'", tier.Name)
 		}
-		if _, err := a.WaitForTierTemplate(t, tier.Spec.ClusterResources.TemplateRef); err != nil {
+		tierTemplateClusterResources, err := a.WaitForTierTemplate(t, tier.Spec.ClusterResources.TemplateRef)
+		if err != nil {
 			return nil, err
 		}
+		// if the tier template supports Tier Template Revisions then let's check those
+		if tierTemplateClusterResources.Spec.TemplateObjects != nil {
+			ttrName, found := tier.Status.Revisions[tierTemplateClusterResources.GetName()]
+			if !found {
+				return nil, fmt.Errorf("missing revision for TierTemplate %s in NSTemplateTier '%s'", tierTemplateClusterResources.GetName(), tier.Name)
+			}
+			if _, err := a.WaitForTierTemplateRevision(t, ttrName); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
this code looks like the same as previous one - why not moving it to a separate method:
```go
func (a *HostAwaitility) checkTTR(t *testing.T, tier *toolchainv1alpha1.NSTemplateTier, tierTemplate *toolchainv1alpha1.TierTemplate) error {
	// if the tier template supports Tier Template Revisions then let's check those
	if tierTemplate.Spec.TemplateObjects != nil {
		ttrName, found := tier.Status.Revisions[tierTemplate.GetName()]
		if !found {
			return fmt.Errorf("missing revision for TierTemplate %s in NSTemplateTier '%s'", tierTemplate.GetName(), tier.Name)
		}
		if _, err := For(t, a.Awaitility, &toolchainv1alpha1.TierTemplateRevision{}).WithNameThat(ttrName); err != nil {
			return err
		}
	}
	return nil
}
```
so you can then call
```go
if err := a.checkTTR(t, tier, tierTemplateClusterResources); err != nil {
	return nil, err
}
```